File: 04/factorial.rb
# factorial.rb, takes an integer n and returns n!
#
# Author: Brian Goins
# Date 2/26/2019 
###################################################

def factorial(n) 
    
    answer = 1
    
    for i in 2..n do
        answer *= i
    end

    return answer

end 


if __FILE__ == $0
    print "\nEnter a number to get its factorial: "
    n = gets.chomp.to_f

    print "\nFactorial = "
    print factorial(n)
    print "\n\n"
end

File: 04/Factorial.java
/* factorial.java, takes an integer n and returns n!
 *
 * Author: Brian Goins
 * Date 2/26/2019 
 ****************************************************/ 



 import java.util.Scanner; 


 class Factorial {
     
    //Main: Prompts user for number and calls factorial function.
    public static void main(String[] args) {

        Scanner keyboard = new Scanner(System.in);
        System.out.print("\nEnter a number to get its factorial: "); 

        double n = keyboard.nextDouble();
        
        System.out.print("\n" + n);
        System.out.printf(" Factorial = %f", factorial(n)); 
        System.out.print("\n\n"); 


        keyboard.close();
    }

    public static double factorial(double n) {
        double answer = 1.0; 

        for(double i = 2; i <= n; ++i) {
            answer *= i; 
        }

        return answer; 
    }

   






 }